src/coreclr/jit/valuenum.cpp

```diff
@@ -3675,12 +3654,14 @@ ValueNum ValueNumStore::VNForExpr(BasicBlock* block, var_types typ)
         loopNum = block->bbNatLoopNum;
     }
 
-    // We always allocate a new, unique VN in this call.
-    // The 'typ' is used to partition the allocation of VNs into different chunks.
-    Chunk*   c                 = GetAllocChunk(typ, CEA_None, loopNum);
-    unsigned offsetWithinChunk = c->AllocVN();
-    ValueNum result            = c->m_baseVN + offsetWithinChunk;
-    return result;
+    // VNForFunc(typ, func, vn) but bypasses looking in the cache
+    //
+    VNDefFunc1Arg fstruct(VNF_MemOpaque, loopNum);
+    Chunk*        c                                                = GetAllocChunk(typ, CEA_Func1);
+    unsigned      offsetWithinChunk                                = c->AllocVN();
+    ValueNum      resultVN                                         = c->m_baseVN + offsetWithinChunk;
+    reinterpret_cast<VNDefFunc1Arg*>(c->m_defs)[offsetWithinChunk] = fstruct;
```

Comment:
This assignment looks odd and is hard to parse.
If it can't be written as a typical assignment, then add a comment with an explanation.

Reply:
You mean the last line there? That is what all of the `VNForFunc` overloads do -- I just "inlined" the unary case here since for these opaque VNs we don't memoize.

Reply:
I see it written here using two lines:

```
ValueNum ValueNumStore::VnForConst(T cnsVal, NumMap* numMap, var_types varType)

        T* chunkDefs                 = reinterpret_cast<T*>(chunk->m_defs);
        chunkDefs[offsetWithinChunk] = cnsVal;
```

Reply:
Ok... might as well change this everywhere it appears.